web-marketplace/src/components/organisms/UserAccountSettings/UserAccountSettings.tsx

```diff
@@ -0,0 +1,147 @@
+        />
+      </div>
+    </div>
+  );
+};
+
+const ConnectField = ({
+  providerName,
+  connect,
+  disconnect,
+  address,
+}: {
+  providerName: string;
+  connect?: () => void;
+  disconnect?: () => void;
+  address?: string;
+}) => {
+  return (
+    <div className="flex flex-row justify-between flex-wrap gap-y-10">
+      <div className="flex flex-col gap-5">
+        <Body size="md" color="info.dark-grey">
+          {providerName}
+        </Body>
+        {address ? (
+          <AddressWidget address={address} />
+        ) : (
+          <Body size="sm">{connect ? 'Disconnected' : 'Connected'}</Body>
+        )}
+      </div>
+      <div>
+        {connect ? (
+          <ContainedButton onClick={connect}>CONNECT</ContainedButton>
+        ) : (
+          <OutlinedButton onClick={disconnect}>
+            <div className="flex flex-row gap-10 items-center">
+              {/* the SVG source for this icon is included inline so that styling with tailwind works */}
+              <svg className="stroke-secondary-main" width="24" height="24" fill="none">
+                <rect width="2" height="17.958" x="6.414" y="19.113" fill="#545555" rx=".25" transform="rotate(-135 6.414 19.113)"/><rect width="2" height="17.958" fill="#545555" rx=".25" transform="scale(1 -1) rotate(45 32.021 11.83)"/>
+              </svg>
+              <div>DISCONNECT</div>
+            </div>
+          </OutlinedButton>
+        )}
+      </div>
+    </div>
+  );
+};
+
+const AddressWidget = ({ address }: { address: string }) => (
+  <div className="flex flex-row gap-5">
+    <Body size="sm">{address}</Body>
+    <CopyButton content={address} tooltip="Copy address" />
+  </div>
+);
+
+const CopyButton = ({content, tooltip}: {content: string, tooltip: string}) => {
+  const [open, setOpen] = React.useState(false);
+  const timerRef = React.useRef(0);
+
+  React.useEffect(() => {
+    return () => clearTimeout(timerRef.current);
+  }, []);
+
+  return <Toast.Provider duration={1000}>
+    <InfoTooltip title={tooltip} arrow placement="top">
+      <svg width="24" height="24" fill="none"
+           onClick={() => {
+             navigator.clipboard.writeText(content);
+             setOpen(false);
+             window.clearTimeout(timerRef.current);
+             timerRef.current = window.setTimeout(() => {
+               setOpen(true);
+             }, 100);
+           }}
+           className="cursor-pointer hover:stroke-grey-300"
+      >
+        <path fill="#000" fillRule="evenodd" d="M19 8v9H9.25a.25.25 0 0 1-.25-.25V3h5v4.25c0 .414.336.75.75.75H19Zm-4-4.586L18.287 7H15V3.414ZM3.25 5H7V1.25A.25.25 0 0 1 7.25 1h8.14c.07 0 .137.03.184.081l5.36 5.847a.25.25 0 0 1 .066.17V18.75a.25.25 0 0 1-.25.25H17v3.75a.25.25 0 0 1-.25.25H3.25a.25.25 0 0 1-.25-.25V5.25A.25.25 0 0 1 3.25 5ZM7 16.75V7H5v14h10v-2H7.25a.25.25 0 0 1-.25-.25v-2Z" clipRule="evenodd"/>
+      </svg>
+    </InfoTooltip>
+    <Toast.Root open={open} onOpenChange={setOpen}
```

Comment:
we already have the `Banner` component so either we should just use it for now or refactor this existing `Banner` component to use `Toast` from radix-ui (instead of MUI `Snackbar`) if we want to move forward with it